### PR TITLE
Querier: remove expensive & low-value debug log line

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
Removes a low-value debug log line that, in one recent example, can logs tens of thousands of times in a short span of time.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the `CONTRIBUTING.md` guide
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by kavirajk — APPROVED** [review]
Sounds reasonable. :+1:

**Review by dannykopping — COMMENTED** [review]
We can also derive this value by looking at the number of keys added to the index cache.

File: pkg/storage/stores/series/index/caching_index_client.go [hunk 1]
```diff
@@ -304,9 +304,6 @@ func (s *cachingIndexClient) cacheStore(ctx context.Context, keys []string, batc
 	hashed := make([]string, 0, len(keys))
 	bufs := make([][]byte, 0, len(batches))
 	for i := range keys {
-		if len(batches[i].Entries) != 0 {
-			level.Debug(logger).Log("msg", "caching index entries", "key", keys[i], "count", len(batches[i].Entries))
```
Comment (MichelHollands) [inline pkg/storage/stores/series/index/caching_index_client.go:308]: The logger variable needs to be removed as well.

Comment (dannykopping) [reply pkg/storage/stores/series/index/caching_index_client.go:308]: Ah right, thanks

**Pull request merged**